osu.Game/Graphics/UserInterface/ShearedToggleButton.cs

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Overlays;
+using osuTK;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class ShearedToggleButton : OsuClickableContainer
+    {
+        public BindableBool Active { get; } = new BindableBool();
+
+        public LocalisableString Text
+        {
+            get => text.Text;
+            set => text.Text = value;
+        }
+
+        private readonly Box background;
+        private readonly OsuSpriteText text;
+
+        private Sample? sampleOff;
+        private Sample? sampleOn;
+
+        [Resolved]
+        private OverlayColourProvider colourProvider { get; set; } = null!;
+
+        public ShearedToggleButton(float shear)
+        {
+            Height = 50;
```

Comment:
I know that this is following by example, but it's very weird that buttons have a hidden requirement of their width being manually set.

How would you feel about something like

```diff
diff --git a/osu.Game/Graphics/UserInterface/ShearedToggleButton.cs b/osu.Game/Graphics/UserInterface/ShearedToggleButton.cs
index acbed29279..c7cb2eb33c 100644
--- a/osu.Game/Graphics/UserInterface/ShearedToggleButton.cs
+++ b/osu.Game/Graphics/UserInterface/ShearedToggleButton.cs
@@ -40,9 +40,10 @@ public LocalisableString Text
         [Resolved]
         private OverlayColourProvider colourProvider { get; set; } = null!;
 
-        public ShearedToggleButton(float shear)
+        public ShearedToggleButton(float shear, float? width = null)
         {
             Height = 50;
+
             Padding = new MarginPadding { Horizontal = shear * 50 };
 
             Content.CornerRadius = 7;
@@ -65,6 +66,16 @@ public ShearedToggleButton(float shear)
                     Shear = new Vector2(-shear, 0)
                 }
             };
+
+            if (width != null)
+            {
+                Width = width.Value;
+            }
+            else
+            {
+                AutoSizeAxes = Axes.X;
+                text.Margin = new MarginPadding { Horizontal = 10 };
+            }
         }
 
         [BackgroundDependencyLoader]

```

I'd probably want the `width` parameter to be first (and have xmldoc and tests), above code is only intended to show intent.

Reply:
Sure, have applied (with slight adjustments).